### PR TITLE
feat: add optional actions to tree component nodes

### PR DESCRIPTION
**Pull request by Kenny806**

### CODE REVIEW

File: src/components/Tree/Node.tsx [hunk 1]
```diff
@@ -25,67 +27,89 @@ type NodeProps = {
...
-                    if (!value) {
-                        setShowNodes(!showNodes);
-                        return;
-                    }
-
-                    if (onClick) {
-                        onClick(activeNodeId === id ? null : id);
-                    }
-                }}
+                onMouseEnter={() => setIsHovered(true)}
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:52]: ()=> setIsHoverd(true) should be move outside of render

File: src/components/Tree/Node.tsx [hunk 2]
```diff
@@ -25,67 +27,89 @@ type NodeProps = {
...
+                onMouseLeave={() => setIsHovered(false)}
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:53]: same as onMouseEnter

File: src/components/Tree/Node.tsx [hunk 3]
```diff
@@ -25,67 +27,89 @@ type NodeProps = {
...
-                <span
-                    data-test-id="toggle"
-                    onClick={(event) => {
-                        event.stopPropagation();
-                        setShowNodes(!showNodes);
+                <a
+                    data-test-id="node-link"
+                    className="tw-flex tw-items-center tw-flex-grow tw-justify-between tw-cursor-pointer"
+                    aria-selected={selected}
+                    onClick={() => {
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:59]: Move outside UI

File: src/components/Tree/Node.tsx [hunk 4]
```diff
@@ -25,67 +27,89 @@ type NodeProps = {
...
-                        selected && "tw-text-black-50",
-                    ])}
-                >
-                    {label}
-                </span>
-            </a>
+                    <div className="tw-flex tw-space-x-2 tw-items-center">
+                        <span
+                            data-test-id="toggle"
+                            onClick={(event) => {
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:73]: move outside UI

File: src/components/Tree/Node.tsx [hunk 5]
```diff
@@ -25,67 +27,89 @@ type NodeProps = {
...
+                <AnimatePresence>
+                    {actions && (isHovered || selected) && (
+                        <motion.div
+                            className="tw-flex tw-space-x-1.5 tw-items-center"
+                            initial={{ width: 0 }}
+                            animate={{ width: "auto" }}
+                            exit={{ width: 0 }}
+                            transition={{ ease: [0.04, 0.62, 0.23, 0.98] }}
+                        >
+                            {actions.map((action) => action)}
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:108]: why not forEach?

File: src/components/Tree/Tree.stories.tsx [hunk 6]
```diff
@@ -1,85 +1,217 @@
...
                             {
-                                id: "1-1-3",
-                                name: "About us",
-                                label: "Page",
-                                value: "https://weare.frontify.com/page/3",
-                                icon: <IconDocument size={IconSize.Size16} />,
+                                id: "dropdown",
+                                size: MenuItemContentSize.Small,
+                                title: "Dropdown",
+                                onClick: () => console.log("dropdown"),
```
Comment (GoranRibic) [inline src/components/Tree/Tree.stories.tsx:46]: remove all console.logs

File: src/components/Tree/Node.tsx (hunk 2 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx]: Done

File: src/components/Tree/Node.tsx (hunk 1 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx]: Done

File: src/components/Tree/Node.tsx (hunk 4 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx:77]: Done

File: src/components/Tree/Node.tsx (hunk 5 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx:112]: TS complains that forEach has no return value, so map seems like a better solution than bending TS to our will :-)

File: src/components/Tree/Node.tsx [hunk 7]
```diff
@@ -25,67 +27,93 @@ type NodeProps = {
...
+    const setHoveredTrue = () => setIsHovered(true);
+    const setHoveredFalse = () => setIsHovered(false);
+    const onNodeClick = () => {
+        if (!value) {
+            setShowNodes(!showNodes);
+            return;
+        }
+
+        if (onClick) {
+            onClick(activeNodeId === id ? null : id);
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:49]: activeNodeId === id can be replaced with selected

File: src/components/Tree/Tree.stories.tsx [hunk 8]
```diff
@@ -1,85 +1,217 @@
...
+
+    return (
+        <Flyout
+            isOpen={open}
+            onOpenChange={(isOpen) => setOpen(isOpen)}
+            legacyFooter={false}
+            hug={false}
+            trigger={
+                <Button
+                    onClick={() => {
```
Comment (GoranRibic) [inline src/components/Tree/Tree.stories.tsx:27]: button onClick move outside return

File: src/components/Tree/Tree.stories.tsx [hunk 9]
```diff
@@ -1,85 +1,217 @@
...
+                        icon: <IconDocument size={IconSize.Size16} />,
+                        label: "Label",
+                        actions: [
+                            <Button
+                                size={ButtonSize.Small}
+                                style={ButtonStyle.Secondary}
+                                icon={<IconPlus size={IconSize.Size12} />}
+                                key="plusIcon"
+                            />,
+                            <ActionsFlyoutComponent key={"actions"} />,
```
Comment (GoranRibic) [inline src/components/Tree/Tree.stories.tsx:201]: no need for {} for key prop

Comment (Kenny806) [reply src/components/Tree/Tree.stories.tsx]: Done

File: src/components/Tree/Tree.stories.tsx (hunk 8 above)
Comment (Kenny806) [reply src/components/Tree/Tree.stories.tsx]: Done

File: src/components/Tree/Node.tsx (hunk 7 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx]: Done

**Review by GoranRibic — APPROVED** [review]

**Pull request merged**